github/search.go

```diff
@@ -172,18 +178,52 @@ func (c CodeResult) String() string {
 // GitHub API docs: https://developer.github.com/v3/search/#search-code
 func (s *SearchService) Code(ctx context.Context, query string, opt *SearchOptions) (*CodeSearchResult, *Response, error) {
 	result := new(CodeSearchResult)
-	resp, err := s.search(ctx, "code", query, opt, result)
+	resp, err := s.search(ctx, "code", &searchParameters{Query: query}, opt, result)
+	return result, resp, err
+}
+
+// LabelsSearchResult represents the result of a code search.
+type LabelsSearchResult struct {
+	Total             *int          `json:"total_count,omitempty"`
+	IncompleteResults *bool         `json:"incomplete_results,omitempty"`
+	Labels            []LabelResult `json:"items,omitempty"`
+}
+
+// LabelResult represents a single search result.
+type LabelResult struct {
+	ID          *int64   `json:"id,omitempty"`
+	URL         *string  `json:"url,omitempty"`
+	Name        *string  `json:"name,omitempty"`
+	Color       *string  `json:"color,omitempty"`
+	Default     *bool    `json:"default,omitempty"`
+	Description *string  `json:"description,omitempty"`
+	Score       *float64 `json:"score,omitempty"`
+}
+
+func (l LabelResult) String() string {
+	return Stringify(l)
+}
+
+// Labels searches labels via various criteria.
+//
+// GitHub API docs: https://developer.github.com/v3/search/#search-labels
+func (s *SearchService) Labels(ctx context.Context, repositoryID string, query string, opt *SearchOptions) (*LabelsSearchResult, *Response, error) {
```

Comment:
Great!

> Should the enum `repositoryID` have been renamed to `repoID` or is OK to keep it more similar to what is in the Gihub API?

Since `searchParameters` is unexported, users won't ever see it (and we can easily change it at any time without a breaking API change). I'm okay with both, so go with whichever you think is nicer.